app/waf/src/services/license.service.ts

```diff
@@ -133,7 +137,16 @@ export class LicenseManager {
 
   // // BIG-IQ ✓<-->x BIG-IP
   // private async licViaBIQ() { }
-  // private async unLicViaBIQ() { }
+  private async unLicViaBIQ(adc: Adc): Promise<void> {
+    let address = adc.management.connection!.ipAddress;
+    let mac = '';
+    for (let net of Object.keys(adc.networks)) {
+      if (adc.networks[net].type === 'mgmt') {
+        mac = adc.management.networks[net].macAddr!;
+      }
+    }
+    await this.biqMgr.revokeLicense(address, mac);
```

Comment:
Without await, `return this.biqMgr.revokeLicense(address, mac);` should be OK.

Reply:
done